File: AdvancePython/Day-2 Constructors and Methods/Task-2.py
# Task 2: parameterized constructure
from enum import Enum


class Type(Enum):
    ACTION = "Action"
    COMEDY = "Comedy"
    ROMANCE = "Romance"
    HORROR = "Horror"


class Movie:
    def __init__(self, name: str, type: Enum, cinema: str) -> None:
        self.name: str = name
        self.type: Enum = type
        self.cinema: str = cinema


movie_data = Movie(name="Avatar: Fire and Ash", type=Type.ACTION, cinema="Hollywood")
print(f"Movie name: {movie_data.name}")
print(f"Movie type: {movie_data.type.value}")
print(f"Movie cinema: {movie_data.cinema}")

File: node.js/Day-2/Task-2.js
// Task 2: parameterized constructure
class Movie {
    constructor(name, type, cinema) {
        this.name = name;
        this.type = type;
        this.cinema = cinema;
    }

    info() {
        console.log(`Movie: ${this.name}, Type: ${this.type}, Cinema: ${this.cinema}`);
    }
}

const movie = new Movie("Iron-man", "Action", "IMAX");
movie.info();